WordPress/Classes/Networking/WordPressClient.swift

```diff
@@ -18,8 +18,8 @@ struct WordPressSite {
 
     init(blog: Blog) throws {
         let url = try ParsedUrl.parse(input: blog.getUrlString())
-        if let account = blog.account {
-            self.init(baseUrl: url, type: .dotCom(authToken: account.authToken))
+        if let account = blog.account, let authToken = account.authToken {
+            self.init(baseUrl: url, type: .dotCom(authToken: authToken))
```

Comment:
As long as a blog has an associated WP.com, it should be using a `.dotCom` client.

Reply:
All `WPAccount` instances should have an `authToken` because they are created after signing into WP.com. We can change the `WordPressClient` to make it support anonymous API requests, but I don't think that's necessary because it won't be used to make anonymous API calls?

Reply:
The issue is that if `account.authToken` is `nil`, it will now fall into the else with `.selfHosted`, which is the logic change. Should it throw an error instead?